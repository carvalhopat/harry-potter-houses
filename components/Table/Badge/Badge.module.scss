.badge {
  text-align: center;
  font-weight: 600;

  & > div {
    padding: 4px;
    border-radius: 4px;
    font-size: 12px;
  }

  .waiting {
    border: 1px solid #939292;
    color: #939292;
  }

  .approved {
    border: 1px solid #17B637;
    color: #17B637;
  }

  .rejected {
    border: 1px solid #E84736;
    color: #E84736;
  }
}
