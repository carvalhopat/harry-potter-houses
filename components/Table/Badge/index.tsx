import PropTypes from 'prop-types';
import styles from './Badge.module.scss'

type TBadgeParams = {
    status: string;
}

function Badge({status}: TBadgeParams){
    return (
        <div className={styles.badge}>
        <div className={styles[status]}>{status}</div>
        </div>
    )
}

Badge.propTypes = {
    status: PropTypes.oneOf(['rejected', 'waiting', 'approved']).isRequired
  };

export default Badge