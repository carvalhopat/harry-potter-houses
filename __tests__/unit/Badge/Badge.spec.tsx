import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Badge from '../../../components/Table/Badge';
import React from 'react';

const badgeStatusList = ['rejected', 'waiting', 'approved']

badgeStatusList.map(badgeStatus => {
    return it(`should render the ${badgeStatus} option in Badge component`, async () => {
        render(<Badge status={badgeStatus} />);
      
        const badge = screen.getByTestId('badge');
      
        expect(badge).toHaveTextContent(badgeStatus);
        expect(badge.classList.contains(badgeStatus)).toBe(true)
        expect(badge).toMatchSnapshot();
      });
})