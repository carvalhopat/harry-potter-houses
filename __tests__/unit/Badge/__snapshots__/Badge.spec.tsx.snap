// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the approved option in Badge component 1`] = `
<div
  class="approved"
  data-testid="badge"
>
  approved
</div>
`;

exports[`should render the rejected option in Badge component 1`] = `
<div
  class="rejected"
  data-testid="badge"
>
  rejected
</div>
`;

exports[`should render the waiting option in Badge component 1`] = `
<div
  class="waiting"
  data-testid="badge"
>
  waiting
</div>
`;
